src/os/bluestore/BlueStore.cc

```diff
@@ -5700,7 +5700,7 @@ int BlueStore::allocate_bluefs_freespace(
     int64_t alloc_len;
     do {
       // hard cap to fit into 32 bits
-      gift = std::min<uint64_t>(size, 1ull << 31);
+      gift = std::min<uint64_t>(size, 1ull << 30);
```

Comment:
doesn't it make sense to lower this even more, e.g.  to 64 or 128 Mb?

Reply:
128 Mb is fine with me:-)

Reply:
The only reason this was there originally was to prevent u32 overflow.  If we actually want a meaningful number there it should be tunable. But.. on a large OSD, i think we should be moving large blocks of space around whenever possible... why make it small?  64MB is about the size of a single SST file in bluefs.

Reply:
As far as I understand the rationale for this commit is to shorten the period of time while globally shared resource(allocator) is exclusively locked. To give a chance for other allocator users to proceed. In this respect the reduction from 4Gb to 1Gb doesn't look very beneficial - chunk allocated per single shot is still  quite large.
Hence the suggestion to make it even shorter....
Surely all the above looks pretty speculative and probably needs some testing/investigation...